prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -317,25 +317,83 @@ int64_t rxr_pkt_req_cq_data(struct rxr_pkt_entry *pkt_entry)
 	return cq_data_hdr->cq_data;
 }
 
-size_t rxr_pkt_req_max_header_size(int pkt_type)
+
+size_t rxr_pkt_req_calc_data_size(struct rxr_ep *ep,
```

Comment:
I see this commit did not remove `rxr_pkt_req_max_data_size` , but introduced a new function `rxr_pkt_req_calc_data_size`. The two functions do exactly same thing. but `rxr_pkt_req_calc_data_size` is accurate and `rxr_pkt_req_max_data_size` has a bug. So I prefer to make the change to `rxr_pkt_req_max_data_size`, instead of introducing a new function.